assets/src/blocks/SecondaryNavigation/SecondaryNavigationFrontend.js

```diff
@@ -0,0 +1,213 @@
+import {useState, useEffect, useRef} from '@wordpress/element';
+import {getHeadingsFromDom} from '../TableOfContents/getHeadingsFromDom';
+import {initializeJustifyContentAdjustment} from './adjustNavWidth';
+
+const makeSecondaryNavigationStickyonScroll = () => {
+  const pageHeader = document.querySelector('.is-pattern-p4-page-header');
+
+  const stickyElement = document.querySelector('.sn-wrapper');
+  const container = document.querySelector('.page-content');
+  const offset = 20;
+
+  window.addEventListener('scroll', () => {
+    const containerRect = container.getBoundingClientRect();
+    const stickyRect = stickyElement.getBoundingClientRect();
+    const pageHeaderRect = pageHeader.getBoundingClientRect();
+
+    if (pageHeaderRect.bottom <= offset && containerRect.bottom > stickyRect.height + offset) {
+      stickyElement.classList.add('stuck');
+    } else {
+      stickyElement.classList.remove('stuck');
+    }
+  });
+};
+
+export const SecondaryNavigationFrontend = ({levels}) => {
+  const [activeLink, setActiveLink] = useState('');
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+  const [currentHeaderLink, setCurrentHeaderLink] = useState('');
+  const [showLeftArrow, setShowLeftArrow] = useState(false);
+  const [showRightArrow, setShowRightArrow] = useState(false);
+  const isClicking = useRef(false);
+  const isManualScroll = useRef(false);
+  const hasLoaded = useRef(false);
+  const navListRef = useRef(null);
+
+  const isMobile = window.innerWidth <= 991;
+
+
+  const headings = getHeadingsFromDom(levels);
+
+  useEffect(() => {
+    makeSecondaryNavigationStickyonScroll();
+    initializeJustifyContentAdjustment();
+  }, []);
+
+  useEffect(() => {
+    const observerOptions = {
+      root: null,
+      rootMargin: '0px',
+      threshold: 1.0,
+    };
+
+    const observerCallback = entries => {
+      if (isClicking.current || isManualScroll.current) {return;}
+
+      // Update the active class for the right heading element on scroll.
+      entries.forEach(entry => {
+        if (!entry.isIntersecting || !hasLoaded.current) {return;}
+
+        const {id, innerText} = entry.target;
+
+        setActiveLink(id);
+        setCurrentHeaderLink(innerText);
+        window.history.replaceState(null, null, `#${id}`);
+
+        // Make active element visible on scroll
+        setTimeout(() => {
+          const navItem = document.querySelector('.secondary-navigation-link.active');
+          navItem?.scrollIntoView({
+            behavior: 'smooth',
+            inline: 'center',
+            block: 'nearest',
+          });
+        }, 0);
+      });
+    };
+
+    const observer = new IntersectionObserver(observerCallback, observerOptions);
+
+    headings.forEach(({anchor}) => {
+      const link = document.getElementById(anchor);
+      if (link) {observer.observe(link);}
+    });
+
+    setTimeout(() => {
```

Comment:
I think that this can be shortened to 1 line: `setTimeout(() => hasLoaded.current = true, 500);`